fvtest/compilertriltest/TernaryTest.cpp

```diff
@@ -0,0 +1,222 @@
+    ASSERT_EQ(0, compileResult) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int64_t (*)(int64_t, int64_t, int64_t, int64_t)>();
+    ASSERT_EQ(param.oracle(param.c1, param.c2, param.v1, param.v2), entry_point(param.c1, param.c2, param.v1, param.v2));
+}
+
+INSTANTIATE_TEST_CASE_P(TernaryTest, Int64TernaryInt64CompareTest,
+        ::testing::Combine(
+            ::testing::ValuesIn(TRTest::const_value_pairs<int64_t, int64_t>()),
+            ::testing::ValuesIn(TRTest::const_value_pairs<int64_t, int64_t>()),
+            ::testing::Values(xternaryOracle<int64_t, int64_t>)));
+
+
+class Int64TernaryDoubleCompareTest : public TernaryTest<double, int64_t> {};
+
+TEST_P(Int64TernaryDoubleCompareTest, TernaryParamI64Double) {
+    auto param = to_struct(GetParam());
+
+    char inputTrees[512] = {0};
+    std::snprintf(inputTrees, sizeof(inputTrees),
+        "(method return=Int64 args=[Double, Double, Int64, Int64]"
+        "  (block"
+        "    (lreturn"
+        "      (lternary"
+        "        (dcmplt"
+        "          (dload parm=0)"
+        "          (dload parm=1))"
+        "        (lload parm=2)"
+        "        (lload parm=3))"
+        ")))");
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    int32_t compileResult = compiler.compile();
+    ASSERT_EQ(0, compileResult) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int64_t (*)(double, double, int64_t, int64_t)>();
+    ASSERT_EQ(param.oracle(param.c1, param.c2, param.v1, param.v2), entry_point(param.c1, param.c2, param.v1, param.v2));
+}
+
+INSTANTIATE_TEST_CASE_P(TernaryTest, Int64TernaryDoubleCompareTest,
+        ::testing::Combine(
+            ::testing::ValuesIn(TRTest::const_value_pairs<double, double>()),
+            ::testing::ValuesIn(TRTest::const_value_pairs<int64_t, int64_t>()),
+            ::testing::Values(xternaryOracle<double, int64_t>)));
+
+class Int32TernaryDoubleCompareTest : public TernaryTest<double, int32_t> {};
+
+TEST_P(Int32TernaryDoubleCompareTest, TernaryParamI32Double) {
+    auto param = to_struct(GetParam());
+
+    char inputTrees[512] = {0};
+    std::snprintf(inputTrees, sizeof(inputTrees),
+        "(method return=Int32 args=[Double, Double, Int32, Int32]"
+        "  (block"
+        "    (ireturn"
+        "      (iternary"
+        "        (dcmplt"
+        "          (dload parm=0)"
+        "          (dload parm=1))"
+        "        (iload parm=2)"
+        "        (iload parm=3))"
+        ")))");
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    int32_t compileResult = compiler.compile();
+
+    ASSERT_EQ(0, compileResult) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
```

Comment:
The formatting here is different than in the other tests (blank lines are in different places).

Reply:
Fixed.